### ISSUE TITLE
Parse Error on Core.hs

### ISSUE THREAD

**Issue by rbharath** [tensorflow/haskell#61]
I'm trying to build the repo using the instructions in `README.md`. However, I'm getting the following error:

```
rbharath@tensorbr0:~/tensorflow-haskell$ sudo stack --docker --docker-image=$IMAGE_NAME test --allow-different-user
tensorflow-core-ops-0.1.0.0: configure (lib)
tensorflow-0.1.0.0: test (suite: FFITest)
tensorflow-0.1.0.0: test (suite: VarIntTest)
tensorflow-core-ops-0.1.0.0: build (lib)
Log files have been written to: /home/rbharath/tensorflow-haskell/.stack-work/logs/
Progress: 2/10
--  While building package tensorflow-core-ops-0.1.0.0 using:
      /home/rbharath/tensorflow-haskell/tensorflow-core-ops/.stack-work/dist/x86_64-linux-dkd1ce2ff9c9560b648268df668d177711/Cabal-1.22.5.0/setup/setup --builddir=.stack-work/dist/x86_64-linux-dkd1ce2ff9c9560b648268df668d177711/Cabal-1.22.5.0 build lib:tensorflow-core-ops --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1
    Logs have been written to: /home/rbharath/tensorflow-haskell/.stack-work/logs/tensorflow-core-ops-0.1.0.0.log

    Configuring tensorflow-core-ops-0.1.0.0...
    Preprocessing library tensorflow-core-ops-0.1.0.0...
    [1 of 1] Compiling TensorFlow.GenOps.Core ( .stack-work/dist/x86_64-linux-dkd1ce2ff9c9560b648268df668d177711/Cabal-1.22.5.0/build/autogen/TensorFlow/GenOps/Core.hs, .stack-work/dist/x86_64-linux-dkd1ce2ff9c9560b648268df668d177711/Cabal-1.22.5.0/build/TensorFlow/GenOps/Core.o )
    
    /home/rbharath/tensorflow-haskell/tensorflow-core-ops/.stack-work/dist/x86_64-linux-dkd1ce2ff9c9560b648268df668d177711/Cabal-1.22.5.0/build/autogen/TensorFlow/GenOps/Core.hs:6766:34:
        parse error on input ‘{’
```
Has anyone seen a similar failure before? I'm on an Ubuntu 16.04 machine.

**Comment by judah** [comment]
Can you look at the source of that (generated) Core.hs file and see which op it's referring to?  We probably need to add more ops to the blacklist here:
https://github.com/tensorflow/haskell/blob/master/tensorflow-core-ops/Setup.hs#L66

In particular, it could be this:
https://github.com/tensorflow/haskell/blob/master/tensorflow-opgen/src/TensorFlow/OpGen.hs#L307

**Comment by rbharath** [comment]
Thanks for the quick response! Good call. I think you got it. Here's the failing line:
```
queueDequeueUpToV2 :: ResourceHandle dtype -- ^ __handle__: The handle to a queue.
                      -> Tensor v2 Data.Int.Int32 -- ^ __n__: The number of tuples to dequeue.
                      -> Build ({{{tensorArg: can't handle heterogeneous lists}}}) -- ^ __components__: One or more tensors that were dequeued as a tuple.
queueDequeueUpToV2 handle n | eqLengthGuard [] =
    buildOp (opDef "QueueDequeueUpToV2")
        handle n
```

What would the fix be here? I'm glad to make a PR if you can point me in the right direction.

**Comment by judah** [comment]
The fix will be to add `"QueueDequeueUpToUpV2"` to `blacklist` here:
https://github.com/tensorflow/haskell/blob/master/tensorflow-core-ops/Setup.hs#L66
There's probably other ops with the same issue; you can search for whether the error message appears in any other places in `Core.hs`.

**Comment by rbharath** [comment]
Made some progress here by adding a few things to `blacklist`. I'm now getting a new error about type-variable `dtype` not being scope:

```
Progress: 10/18
--  While building package tensorflow-core-ops-0.1.0.0 using:
      /home/rbharath/tensorflow-haskell/tensorflow-core-ops/.stack-work/dist/x86_64-linux-dkd1ce2ff9c9560b648268df668d177711/Cabal-1.22.5.0/setup/setup --builddir=.stack-work/dist/x86_64-linux-dkd1ce2ff9c9560b648268df668d177711/Cabal-1.22.5.0 build lib:tensorflow-core-ops --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1
    Logs have been written to: /home/rbharath/tensorflow-haskell/.stack-work/logs/tensorflow-core-ops-0.1.0.0.log

    [1 of 2] Compiling Main             ( /home/rbharath/tensorflow-haskell/tensorflow-core-ops/Setup.hs, /home/rbharath/tensorflow-haskell/tensorflow-core-ops/.stack-work/dist/x86_64-linux-dkd1ce2ff9c9560b648268df668d177711/Cabal-1.22.5.0/setup/Main.o )
    Linking /home/rbharath/tensorflow-haskell/tensorflow-core-ops/.stack-work/dist/x86_64-linux-dkd1ce2ff9c9560b648268df668d177711/Cabal-1.22.5.0/setup/setup ...
    Preprocessing library tensorflow-core-ops-0.1.0.0...
    [1 of 1] Compiling TensorFlow.GenOps.Core ( .stack-work/dist/x86_64-linux-dkd1ce2ff9c9560b648268df668d177711/Cabal-1.22.5.0/build/autogen/TensorFlow/GenOps/Core.hs, .stack-work/dist/x86_64-linux-dkd1ce2ff9c9560b648268df668d177711/Cabal-1.22.5.0/build/TensorFlow/GenOps/Core.o )

    /home/rbharath/tensorflow-haskell/tensorflow-core-ops/.stack-work/dist/x86_64-linux-dkd1ce2ff9c9560b648268df668d177711/Cabal-1.22.5.0/build/autogen/TensorFlow/GenOps/Core.hs:8981:41:
        Not in scope: type variable ‘dtype’

    /home/rbharath/tensorflow-haskell/tensorflow-core-ops/.stack-work/dist/x86_64-linux-dkd1ce2ff9c9560b648268df668d177711/Cabal-1.22.5.0/build/autogen/TensorFlow/GenOps/Core.hs:8982:44:
        Not in scope: type variable ‘dtype’
...
```

(The `dtype` errors happen in a bunch of places. Only copied over first two failures). Any thoughts on potential fixes?

**Comment by judah** [comment]
Ah, unfortunately it looks like we also need to blacklist some more ops to work around an issue with some ResourceHandle ops.  I've sent out #62.

**Comment by rbharath** [comment]
Nice fix! With #62 checked out locally, I'm able to build successfully. I'll close this issue once #62 is merged in then.
